api/infrastructure/parallelcluster-api.yaml

```diff
@@ -142,6 +149,412 @@ Resources:
               - '-'
               - [!Select [2, !Split ['/', !Ref EcrImage]], !Select [3, !Split ['/', !Ref EcrImage]]]
 
+  ParallelClusterFunctionPolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      Roles:
+      - !Ref ParallelClusterFunctionRole
```

Comment:
it can be done as part of a separate PR but we need to offer a parameter to override this role

Reply:
What do you mean? The role is the role implicitly generated by SAM for the `ParallelClusterFunction` `AWS::Serverless::Function` resource

Reply:
As discussed, I will add a parameter to allow the customer to specify their own role